planner/core/find_best_task.go

```diff
@@ -1588,27 +1590,25 @@ func (is *PhysicalIndexScan) initSchema(idxExprCols []*expression.Column, isDoub
 		}
 	}
 
-	if isDoubleRead {
```

Comment:
Now this is necessary. The origin reason: if we have to filter the partitions in TiDB, we have to tell tikv to return ExtraPid. So that we can filter by the returned Pid.